doc/architectural/CBMC-developer-guide.md

```diff
@@ -0,0 +1,507 @@
+\ingroup module_hidden 
+\page cbmc-developer-guide CBMC Developer Guide
+
+\author Martin Brain, Peter Schrammel, César Rodríguez, Owen Jones
+
+\tableofcontents
+
+\section section-compilation-and-development Compilation and Development
+
+## Makefiles ##
+
+First off, read the \ref cbmc-user-manual "CBMC User Manual". It describes
+how to get, build and use CBMC. This document covers the
+internals of the system and how to get started on development.
+
+## CMake files ##
+
+To be documented.
+
+## Personal configuration: config.inc, macro DEBUG ##
+
+To be documented.
+
+## Generating doxygen documentation ##
+
```

Comment:
If the latest version of this is going to be available in http://cprover.diffblue.com/ it should be worth mentioning it here.